### PR TITLE
fix(bridge): don't log err (contains the x-token), only log err.message (#6047)

### LINKED ISSUE #6047: shipyard-controller: Monitoring detects 500 errors when a service is created which already exists
**Environment**

* Client OS (e.g., Linux, macOS, Windows): any
* Keptn Version (`keptn version`): 0.10.0
* Kubernetes Cloud Provider (e.g., GKE, AKS): any
* Kubernetes version (`kubectl version`): any

**Affected Component**

* [ ] Approval
* [x] Bridge
* [ ] CLI
* [ ] Datastore
* [ ] Docs
* [ ] Distributor / Event Broker
* [ ] Helm
* [ ] Istio
* [ ] Jmeter
* [ ] Kubernetes Integration
* [ ] Openshift Integration
* [ ] REST API
* [ ] Webhook
* [x] shipyard-controller

**Describe the bug**
In the logs of the Bridge server, I see the following error: 
```
Error: could not create service dynatrace in stage quality-gate of project dynatrace: Service already exists
    at createError (/usr/src/app/server/node_modules/axios/lib/core/createError.js:16:15)
    at settle (/usr/src/app/server/node_modules/axios/lib/core/settle.js:17:12)
    at IncomingMessage.handleStreamEnd (/usr/src/app/server/node_modules/axios/lib/adapters/http.js:269:11)
    at IncomingMessage.emit (events.js:412:35)
    at endReadableNT (internal/streams/readable.js:1334:12)
    at processTicksAndRejections (internal/process/task_queues.js:82:21) {
  config: {
    url: 'http://api-gateway-nginx/api/controlPlane/v1/project/dynatrace/service',
    method: 'post',
    data: '{"serviceName":"dynatrace"}',
    headers: {
      Accept: 'application/json, text/plain, */*',
      'Content-Type': 'application/json',
      'x-token': '',
      'User-Agent': 'axios/0.21.4',
      'Content-Length': 27
    },
    transformRequest: [ [Function: transformRequest] ],
    transformResponse: [ [Function: transformResponse] ],
    timeout: 0,
    adapter: [Function: httpAdapter],
    xsrfCookieName: 'XSRF-TOKEN',
    xsrfHeaderName: 'X-XSRF-TOKEN',
    maxContentLength: -1,
    maxBodyLength: -1,
    httpsAgent: Agent {
      _events: [Object: null prototype],
      _eventsCount: 2,
      _maxListeners: undefined,
      defaultPort: 443,
      protocol: 'https:',
      options: [Object],
      requests: {},
      sockets: {},
      freeSockets: {},
      keepAliveMsecs: 1000,
      keepAlive: false,
      maxSockets: Infinity,
      maxFreeSockets: 256,
      scheduling: 'lifo',
      maxTotalSockets: Infinity,
      totalSocketCount: 0,
      maxCachedSessions: 100,
      _sessionCache: [Object],
      [Symbol(kCapture)]: false
    },
    validateStatus: [Function: validateStatus],
    transitional: {
      silentJSONParsing: true,
      forcedJSONParsing: true,
      clarifyTimeoutError: false
    }
  },
  request: <ref *1> ClientRequest {
    _events: [Object: null prototype] {
      abort: [Function (anonymous)],
      aborted: [Function (anonymous)],
      connect: [Function (anonymous)],
      error: [Function (anonymous)],
      socket: [Function (anonymous)],
      timeout: [Function (anonymous)],
      prefinish: [Function: requestOnPrefinish]
    },
    _eventsCount: 7,
    _maxListeners: undefined,
    outputData: [],
    outputSize: 0,
    writable: true,
    destroyed: false,
    _last: true,
    chunkedEncoding: false,
    shouldKeepAlive: false,
    _defaultKeepAlive: true,
    useChunkedEncodingByDefault: true,
    sendDate: false,
    _removedConnection: false,
    _removedContLen: false,
    _removedTE: false,
    _contentLength: null,
    _hasBody: true,
    _trailer: '',
    finished: true,
    _headerSent: true,
    socket: Socket {
      connecting: false,
      _hadError: false,
      _parent: null,
      _host: 'api-gateway-nginx',
      _readableState: [ReadableState],
      _events: [Object: null prototype],
      _eventsCount: 7,
      _maxListeners: undefined,
      _writableState: [WritableState],
      allowHalfOpen: false,
      _sockname: null,
      _pendingData: null,
      _pendingEncoding: '',
      server: null,
      _server: null,
      parser: null,
      _httpMessage: [Circular *1],
      [Symbol(async_id_symbol)]: 530446,
      [Symbol(kHandle)]: [TCP],
      [Symbol(kSetNoDelay)]: false,
      [Symbol(lastWriteQueueSize)]: 0,
      [Symbol(timeout)]: null,
      [Symbol(kBuffer)]: null,
      [Symbol(kBufferCb)]: null,
      [Symbol(kBufferGen)]: null,
      [Symbol(kCapture)]: false,
      [Symbol(kBytesRead)]: 0,
      [Symbol(kBytesWritten)]: 0,
      [Symbol(RequestTimeout)]: undefined
    },
    _header: 'POST /api/controlPlane/v1/project/dynatrace/service HTTP/1.1\r\n' +
      'Accept: application/json, text/plain, */*\r\n' +
      'Content-Type: application/json\r\n' +
      'x-token: \r\n' +
      'User-Agent: axios/0.21.4\r\n' +
      'Content-Length: 27\r\n' +
      'Host: api-gateway-nginx\r\n' +
      'Connection: close\r\n' +
      '\r\n',
    _keepAliveTimeout: 0,
    _onPendingData: [Function: noopPendingOutput],
    agent: Agent {
      _events: [Object: null prototype],
      _eventsCount: 2,
      _maxListeners: undefined,
      defaultPort: 80,
      protocol: 'http:',
      options: [Object],
      requests: {},
      sockets: [Object],
      freeSockets: {},
      keepAliveMsecs: 1000,
      keepAlive: false,
      maxSockets: Infinity,
      maxFreeSockets: 256,
      scheduling: 'lifo',
      maxTotalSockets: Infinity,
      totalSocketCount: 1,
      [Symbol(kCapture)]: false
    },
    socketPath: undefined,
    method: 'POST',
    maxHeaderSize: undefined,
    insecureHTTPParser: undefined,
    path: '/api/controlPlane/v1/project/dynatrace/service',
    _ended: true,
    res: IncomingMessage {
      _readableState: [ReadableState],
      _events: [Object: null prototype],
      _eventsCount: 3,
      _maxListeners: undefined,
      socket: [Socket],
      httpVersionMajor: 1,
      httpVersionMinor: 1,
      httpVersion: '1.1',
      complete: true,
      headers: [Object],
      rawHeaders: [Array],
      trailers: {},
      rawTrailers: [],
      aborted: false,
      upgrade: false,
      url: '',
      method: null,
      statusCode: 500,
      statusMessage: 'Internal Server Error',
      client: [Socket],
      _consuming: false,
      _dumped: false,
      req: [Circular *1],
      responseUrl: 'http://api-gateway-nginx/api/controlPlane/v1/project/dynatrace/service',
      redirects: [],
      [Symbol(kCapture)]: false,
      [Symbol(RequestTimeout)]: undefined
    },
    aborted: false,
    timeoutCb: null,
    upgradeOrConnect: false,
    parser: null,
    maxHeadersCount: null,
    reusedSocket: false,
    host: 'api-gateway-nginx',
    protocol: 'http:',
    _redirectable: Writable {
      _writableState: [WritableState],
      _events: [Object: null prototype],
      _eventsCount: 2,
      _maxListeners: undefined,
      _options: [Object],
      _ended: true,
      _ending: true,
      _redirectCount: 0,
      _redirects: [],
      _requestBodyLength: 27,
      _requestBodyBuffers: [],
      _onNativeResponse: [Function (anonymous)],
      _currentRequest: [Circular *1],
      _currentUrl: 'http://api-gateway-nginx/api/controlPlane/v1/project/dynatrace/service',
      [Symbol(kCapture)]: false
    },
    [Symbol(kCapture)]: false,
    [Symbol(kNeedDrain)]: false,
    [Symbol(corked)]: 0,
    [Symbol(kOutHeaders)]: [Object: null prototype] {
      accept: [Array],
      'content-type': [Array],
      'x-token': [Array],
      'user-agent': [Array],
      'content-length': [Array],
      host: [Array]
    }
  },
  response: {
    status: 500,
    statusText: 'Internal Server Error',
    headers: {
      server: 'nginx/1.21.3',
      date: 'Thu, 18 Nov 2021 07:46:32 GMT',
      'content-type': 'application/json; charset=utf-8',
      'server-timing': 'dtSInfo;desc="0", dtRpid;desc="-523479200"',
      'set-cookie': [Array],
      'content-length': '126',
      'cache-control': 'no-cache, no-store, max-age=0, must-revalidate',
      pragma: 'no-cache',
      expires: '0',
      'x-content-type-options': 'nosniff',
      'x-frame-options': 'DENY',
      'x-xss-protection': '1 ; mode=block',
      'referrer-policy': 'no-referrer',
      connection: 'close'
    },
    config: {
      url: 'http://api-gateway-nginx/api/controlPlane/v1/project/dynatrace/service',
      method: 'post',
      data: '{"serviceName":"dynatrace"}',
      headers: [Object],
      transformRequest: [Array],
      transformResponse: [Array],
      timeout: 0,
      adapter: [Function: httpAdapter],
      xsrfCookieName: 'XSRF-TOKEN',
      xsrfHeaderName: 'X-XSRF-TOKEN',
      maxContentLength: -1,
      maxBodyLength: -1,
      httpsAgent: [Agent],
      validateStatus: [Function: validateStatus],
      transitional: [Object]
    },
    request: <ref *1> ClientRequest {
      _events: [Object: null prototype],
      _eventsCount: 7,
      _maxListeners: undefined,
      outputData: [],
      outputSize: 0,
      writable: true,
      destroyed: false,
      _last: true,
      chunkedEncoding: false,
      shouldKeepAlive: false,
      _defaultKeepAlive: true,
      useChunkedEncodingByDefault: true,
      sendDate: false,
      _removedConnection: false,
      _removedContLen: false,
      _removedTE: false,
      _contentLength: null,
      _hasBody: true,
      _trailer: '',
      finished: true,
      _headerSent: true,
      socket: [Socket],
      _header: 'POST /api/controlPlane/v1/project/dynatrace/service HTTP/1.1\r\n' +
        'Accept: application/json, text/plain, */*\r\n' +
        'Content-Type: application/json\r\n' +
        'x-token: \r\n' +
        'User-Agent: axios/0.21.4\r\n' +
        'Content-Length: 27\r\n' +
        'Host: api-gateway-nginx\r\n' +
        'Connection: close\r\n' +
        '\r\n',
      _keepAliveTimeout: 0,
      _onPendingData: [Function: noopPendingOutput],
      agent: [Agent],
      socketPath: undefined,
      method: 'POST',
      maxHeaderSize: undefined,
      insecureHTTPParser: undefined,
      path: '/api/controlPlane/v1/project/dynatrace/service',
      _ended: true,
      res: [IncomingMessage],
      aborted: false,
      timeoutCb: null,
      upgradeOrConnect: false,
      parser: null,
      maxHeadersCount: null,
      reusedSocket: false,
      host: 'api-gateway-nginx',
      protocol: 'http:',
      _redirectable: [Writable],
      [Symbol(kCapture)]: false,
      [Symbol(kNeedDrain)]: false,
      [Symbol(corked)]: 0,
      [Symbol(kOutHeaders)]: [Object: null prototype]
    },
    data: {
      code: 500,
      message: 'could not create service dynatrace in stage quality-gate of project dynatrace: Service already exists'
    }
  },
  isAxiosError: true,
  toJSON: [Function: toJSON]
}
```

**Current behavior**
- If a service already exists, no 500 error should be thrown. It must be a 409
- The Bridge must not log the x-token (I manually removed it above)

### PR DESCRIPTION
**Pull request by ermin-muratovic**
Signed-off-by: ermin.muratovic <ermin.muratovic@gmail.com>

## This PR
- adjusts the log to not show the x-token when an error occurs

### Related Issues
Fixes #6047

### CODE REVIEW

**Review by thisthat — APPROVED** [review]

File: bridge/server/app.ts [hunk 1]
```diff
@@ -192,7 +192,7 @@ async function init(): Promise<Express> {
     }
 
     res.status(err.response?.status || 500).send(err.message);
-    console.error(err);
+    console.error(`[ERROR] ${err.message}`);
```
Comment (Kirdock) [inline bridge/server/app.ts:195]: I would also log the requested URL and status code here if they exist. Without this, it is not possible to see where this error came from

**Review by Kirdock — CHANGES_REQUESTED** [review]

File: bridge/server/app.ts (hunk 1 above)
Comment (Kirdock) [reply bridge/server/app.ts:195]: The whole stack trace is also lost. We should differentiate if the error is an instance of Error or not.

Comment (Kirdock) [reply bridge/server/app.ts:195]: I would also change the err type of `any` to `Error | AxiosError`

Comment (ermin-muratovic) [reply bridge/server/app.ts]: Sure, we can differentiate `Error` which might happen in the application and `AxiosError` which come from API calls.

File: bridge/server/app.ts [hunk 2]
```diff
@@ -179,20 +180,26 @@ async function init(): Promise<Express> {
 
   // error handler
   // eslint-disable-next-line @typescript-eslint/no-explicit-any,@typescript-eslint/no-unused-vars
```
Comment (Kirdock) [inline bridge/server/app.ts:182]: now we don't need to disable the rule `@typescript-eslint/no-explicit-any,`

**Review by Kirdock — CHANGES_REQUESTED** [review]

**Review by Kirdock — APPROVED** [review]

**Pull request merged**